core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -77,6 +77,9 @@ abstract class AbstractFetcherThread(name: String,
       partitionMapCond.signalAll()
     }
     awaitShutdown()
+
+    fetcherStats.unregister()
+    fetcherLagStats.unregister()
```

Comment:
We could grab the lock again for consistency I guess, but the metrics removal seems safe as it is.